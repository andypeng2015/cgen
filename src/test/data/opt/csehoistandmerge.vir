module foo

export function w $foo(w %x) {
@start:
  %c = sge.w %x, 0_w
  br %c, @pos, @neg
@pos:
  %a = div.w %x, 4_w
  %r = add.w %a, 1_w
  jmp @end
@neg:
  %a = div.w %x, 4_w
  %r = add.w %a, 1_w
  jmp @end
@end:
  ret %r
}
