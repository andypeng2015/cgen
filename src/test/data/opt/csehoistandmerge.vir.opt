module foo

export function w $foo(w %x) {
@2:
  %6 = slt.w %x, 0x0_w ; @17
  %7 = add.w %x, 0x3_w ; @18
  %8 = sel.w %6, %7, %x ; @19
  %9 = asr.w %8, 0x2_w ; @20
  %10 = add.w %9, 0x1_w ; @21
  ret %10
}
