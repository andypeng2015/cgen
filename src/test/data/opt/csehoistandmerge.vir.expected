module foo

export function w $foo(w %x) {
@0:
  %3 = slt.w %x, 0x0_w ; @12
  %4 = add.w %x, 0x3_w ; @13
  %2 = sel.w %3, %4, %x ; @11
  %1 = asr.w %2, 0x2_w ; @10
  %0 = add.w %1, 0x1_w ; @9
  ret %0
}
